### PR TITLE
adding train_test_split functionality (incomplete)

### PR DESCRIPTION
**Pull request by hgupta18**

### CODE REVIEW

**Review by johannes-kk — APPROVED** [review]
Offloads shuffling to `sample`, but we can implement it there instead, per #71.
